File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/template_method/lab/CreditCardPurchase.java
// CreditCardPurchase.java

public class CreditCardPurchase extends Purchase
{
	public void perform()
	{
		selectItem();
		bringItemToRegister();
		payViaCreditCard();
	}

	private void selectItem()
	{
	}

	private void bringItemToRegister()
	{
	}

	private void payViaCreditCard()
	{
		System.out.println("Paying via credit card.");
	}
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_matlab_labs_and_solutions/template_method/lab/CreditCardPurchase.m
% CreditCardPurchase.m

classdef CreditCardPurchase < Purchase
	methods
		function perform(theCreditCardPurchase)
			theCreditCardPurchase.selectItem;
			theCreditCardPurchase.bringItemToRegister;
			theCreditCardPurchase.payViaCreditCard;
		end
	end

	methods (Access = private)
		function selectItem(theCreditCardPurchase)
		end

		function bringItemToRegister(theCreditCardPurchase)
		end

		function payViaCreditCard(theCreditCardPurchase)
			fprintf('Paying via credit card.\n');
		end
	end
end
